python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -529,15 +538,17 @@ def __init__(cls, name, bases, this_dict):
 
         # all Ganga classes must have (even empty) schema
         if cls._schema is None:
-            cls._schema = Schema.Schema(None, None)
+            cls._schema = Schema(None, None)
 
         this_schema = cls._schema
 
+        cls.__slots__ = ('__dict__', '_proxyObject')
```

Comment:
The way that `__slots__` improves the memory usage is by disabling the automatic creation of a `__dict_-` per object and instead does them as descriptors. As soon as you allow `__dict__` to be in `__slots__` it undoes that and gives every object its dictionary back.